source/neuropod/internal/neuropod_tensor.cc

```diff
@@ -23,6 +23,17 @@ namespace neuropod
 namespace
 {
 
+size_t compute_num_elements(const std::vector<int64_t> &dims)
+{
+    // Get the number of elements in the tensor by multiplying all the dims together
+    return static_cast<size_t>(std::accumulate(dims.begin(), dims.end(), 1, std::multiplies<>()));
```

Comment:
#include <functional>   // std:: multiplies
#include <numeric>      // std::accumulate

Reply:
btw, it may overflow theoretically but maybe at this moment we may assume that dims are valid for this operation